redwood-protocol-guest/src/commonMain/kotlin/app/cash/redwood/protocol/guest/ProtocolBridge.kt

```diff
@@ -34,14 +34,14 @@ public interface ProtocolBridge : EventSink {
    * The root of the widget tree onto which [widgetSystem]-produced widgets can be added. Changes to
    * this instance are recorded as changes to [Id.Root] and [ChildrenTag.Root].
    */
-  public val root: Widget.Children<Nothing>
+  public val root: Widget.Children<Unit>
```

Comment:
Actually looking at it now, I think we can pass the `Widget<T>` to the `WidgetSystem` and have it responsible for unwrapping the value. The protocol-based, guest-side implementation is empty, so that allows us to keep `Nothing`.

This is only possible because I dropped the ability to return a wrapping native UI node. So when that gets added back we might need to make this change again. Going to walk this back and make a smaller diff!

Reply:
Nope this won't work 😢 . The `WidgetSystem` subtype is the same for both Redwood and Treehouse guest code. It's only the factories that change. Oh well!